packages/cheerio-crawler/src/internals/cheerio-crawler.ts

```diff
@@ -850,26 +847,24 @@ export class CheerioCrawler<JSONData = unknown> extends BasicCrawler<
      */
     protected async _requestFunction({ request, session, proxyUrl, requestAsBrowserOptions }: RequestFunctionOptions): Promise<IncomingMessage> {
         const opts = this._getRequestOptions(request, session, proxyUrl, requestAsBrowserOptions);
-        let responseWithStream: IncomingMessage;
 
         try {
-            responseWithStream = await this._requestAsBrowser(opts);
+            return await this._requestAsBrowser(opts);
         } catch (e) {
             if (e instanceof TimeoutError) {
                 this._handleRequestTimeout(session);
-            } else {
-                throw e;
+                return undefined as unknown as IncomingMessage;
```

Comment:
ouch :D looking at the `_handleRequestTimeout` implementation, i would vote to inline it so TS understands we throw in this branch.